packages/webapp/src/components/Form/CompositionInputs/NumberInputWithSelect.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import clsx from 'clsx';
+import { useTranslation } from 'react-i18next';
+import { Controller } from 'react-hook-form';
+import NumberInput from '../NumberInput';
+import ReactSelect from '../ReactSelect';
+import useReactSelectStyles from '../Unit/useReactSelectStyles';
+import InputBaseLabel from '../InputBase/InputBaseLabel';
+import styles from './styles.module.scss';
+
+type NumberInputWithSelectProps = {
+  control: any;
```

Comment:
Does the Control type from react-hook-form work here?

Reply:
It wasn't simple at all 😢, but I used the Control type. I'm not sure if I did it correctly though 😭

Reply:
It looks correct to me! But also looks unnecessary and a headache to maintain so I made some comments on improvement in the review.